### PR TITLE
Bugfix/label image fixes

### PR DESCRIPTION
**Pull request by robert-kalmar**
This PR fixes following issues with label_image TFLite example: 

1. Behavior of _loop count_ (-c) and _warm_up count_ (-w) command line switch:
Without this patch the warmup_count depends on loop_counts. If the loop_count is less than 2, no warmup loop is performed. E.g. this call will make no warm up loops, however explicitly requested:
`$ ./label_image -c1 -w2`
This call will make 2 Interptetter::Invoke() loops and 2 warm up calls:
`$ ./label_image -c2 -w2`
With this patch, both commands will do 1 resp. 2 Intepretter::Invoke loops and 2 warm up loops, as intuitively expected. 

2. Help message printing for label_image:
There are two paths from where the help message is printed, with different content:
`$ ./label_image -h` will print the help message defined in display_usage(). Moreover it does not even recognize the -h option and getopt print Unrecognized switch error (-h is missing in getopt optstring).
`$ ./label_image --help` prints the help message provided by delegate registrar. 
Switches in both help messages are valid, so none of the help messages provides a complete usage information. 
This patch fix the issue and unites the messages.

### CODE REVIEW

**Comment by MeghnaNatraj** [comment]
@multiverse-tf I have reviewed the change and it looks good. Could you review 1 of the 2 changes in this commit again? - https://github.com/tensorflow/tensorflow/pull/51854/commits/d93205254d9fedab091fddc97ec9731ebac5ab31 (I'm not sure if this is required)

**Comment by robert-kalmar** [comment]
> @multiverse-tf I have reviewed the change and it looks good. Could you review 1 of the 2 changes in this commit again? - [d932052](https://github.com/tensorflow/tensorflow/commit/d93205254d9fedab091fddc97ec9731ebac5ab31) (I'm not sure if this is required)

**Comment by robert-kalmar** [comment]
> @multiverse-tf I have reviewed the change and it looks good. Could you review 1 of the 2 changes in this commit again? - [d932052](https://github.com/tensorflow/tensorflow/commit/d93205254d9fedab091fddc97ec9731ebac5ab31) (I'm not sure if this is required)
@MeghnaNatraj which change you mean? The [d932052](https://github.com/tensorflow/tensorflow/commit/d93205254d9fedab091fddc97ec9731ebac5ab31) contains multiple changes (there are changes in 5 files). Maybe I can clarify them.

**Comment by gbaned** [comment]
@multiverse-tf Can you please review this PR ? Thanks!

File: tensorflow/lite/tools/delegates/delegate_provider.cc [hunk 1]
```diff
@@ -32,6 +32,18 @@ void ProvidedDelegateList::AppendCmdlineFlags(std::vector<Flag>* flags) const {
   }
 }
 
+void ProvidedDelegateList::RemoveCmdlineFlag(std::vector<Flag>& flags,
```
Comment (multiverse-tf) [inline tensorflow/lite/tools/delegates/delegate_provider.cc:35]: Move this function as part of class "Flags" considering the functionality has nothing to do with ProvidedDelegateList?

**Comment by multiverse-tf** [comment]
Sorry for the late review, and thanks for the contribution! Overall, it looks good to me except one small comment.

File: tensorflow/lite/tools/delegates/delegate_provider.cc (hunk 1 above)
Comment (robert-kalmar) [reply tensorflow/lite/tools/delegates/delegate_provider.cc:35]: The functionality is provided in `ProvidedDelegateList` class as the counterpart of `AppendCmdlineFlags` method (already in the class). The `Flags` class currently serves only to parse/print:  
```
class Flags {
 public:
  static bool Parse(int* argc, const char** argv,
                    const std::vector<Flag>& flag_list);

  static std::string Usage(const std::string& cmdline,
                           const std::vector<Flag>& flag_list);

  static std::string ArgsToString(int argc, const char** argv);
};
```

Comment (multiverse-tf) [reply tensorflow/lite/tools/delegates/delegate_provider.cc:35]: This makes sense. Btw, to be consistent with the declaration of AppendCmdlineFlags, change "std::vector<Flag>& flags" to "std::vector<Flag>* flags" or vice versa?

Comment (robert-kalmar) [reply tensorflow/lite/tools/delegates/delegate_provider.cc:35]: Make sense. For consistency replaced the pointer to reference in `AppendCmdlineFlags`. The pointer is not necessary. Moreover this way it is not necessary to check the pointer validity (i.e. != nullptr).

Comment (multiverse-tf) [reply tensorflow/lite/tools/delegates/delegate_provider.cc:35]: Great! Thx!

**Review by terryheo — APPROVED** [review]
LGTM

**Pull request merged**